### PR TITLE
GT-350 Backup lifetime - remove Backup once its lifetime has been reached

### PR DESCRIPTION
**Pull request by jwierzbo**

### CODE REVIEW

File: pkg/apis/backup/v1/backup_spec.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ type ArangoBackupSpec struct {
 	PolicyName *string `json:"policyName,omitempty"`
 
 	Backoff *ArangoBackupSpecBackOff `json:"backoff,omitempty"`
+
```
Comment (ajanikow) [inline pkg/apis/backup/v1/backup_spec.go:38]: Synchronize

File: pkg/handlers/backup/state_ready.go [hunk 2]
```diff
@@ -56,6 +58,25 @@ func stateReadyHandler(h *handler, backup *backupApi.ArangoBackup) (*backupApi.A
 		)
 	}
 
+	if backup.Spec.Lifetime != "" {
```
Comment (ajanikow) [inline pkg/handlers/backup/state_ready.go:61]: Any test?

File: pkg/apis/backup/v1/backup_spec.go [hunk 3]
```diff
@@ -35,6 +35,9 @@ type ArangoBackupSpec struct {
...
+	// Lifetime is the time after which the backup will be deleted. Format: "1.5h" or "2h45m".
+	Lifetime string `json:"lifetime,omitempty"`
```
Comment (ajanikow) [inline pkg/apis/backup/v1/backup_spec.go:40]: Try meta.Duration instead of custom parser

**Review by ajanikow — CHANGES_REQUESTED** [review]

**Comment by jwierzbo** [comment]
Tests: https://github.com/arangodb/kube-arangodb-test/pull/164/files

**Comment by jwierzbo** [comment]
http://a449ed7c695a44a0faa7c64adce736b0-340255436.eu-central-1.elb.amazonaws.com/job/OperatorTestRun/923/parameters/

**Review by ajanikow — APPROVED** [review]

**Pull request merged**